helix-loader/src/lib.rs

```diff
@@ -281,21 +299,3 @@ mod merge_toml_tests {
         )
     }
 }
-
-/// Finds the current workspace folder.
-/// Used as a ceiling dir for LSP root resolution, the filepicker and potentially as a future filewatching root
-///
-/// This function starts searching the FS upward from the CWD
-/// and returns the first directory that contains either `.git` or `.helix`.
-/// If no workspace was found returns (CWD, true).
-/// Otherwise (workspace, false) is returned
-pub fn find_workspace() -> (PathBuf, bool) {
-    let current_dir = std::env::current_dir().expect("unable to determine current directory");
-    for ancestor in current_dir.ancestors() {
-        if ancestor.join(".git").exists() || ancestor.join(".helix").exists() {
-            return (ancestor.to_owned(), false);
-        }
-    }
-
-    (current_dir, true)
-}
```

Comment:
Why the change here?

Reply:
Clippy says the test items should avoid be declared after testing mod: https://rust-lang.github.io/rust-clippy/master/index.html#items_after_test_module

So I just move the testing mod down, but git diff shows the changes on `find_workspace` instead.

```rust
warning: items were found after the testing module
   --> helix-loader\src\lib.rs:213:1
    |
213 | / mod merge_toml_tests {
214 | |     use std::str;
215 | |
216 | |     use super::merge_toml_values;
...   |
300 | |     (current_dir, true)
301 | | }
    | |_^
    |
    = help: move the items to before the testing module was defined
    = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#items_after_test_module
    = note: `#[warn(clippy::items_after_test_module)]` on by default
```

Reply:
Oh, I think the reason why I meet clippy complaints is that I use a rustc/cargo in a higher verion.

`items_after_test_module` hasn't been in [clippy1.65](https://rust-lang.github.io/rust-clippy/rust-1.65.0/) yet.

So the Github action doesn't catch them for now.

If this PR is not appropriate for now, feel free to close.

Reply:
Yeah all these are future lints that don't trigger in current version yet but it's good to fix them ahead of time 👍🏻